docs/pages/enroll-resources/workload-identity/aws-oidc-federation.mdx

```diff
@@ -0,0 +1,431 @@
+version: v6
+metadata:
+  name: my-workload-aws
+spec:
+  allow:
+    spiffe:
+      - path: /svc/example-service
+```
+
+Replace:
+
+- `my-workload-aws` with a descriptive name for the role.
+- `/svc/example-service` with the path part of the SPIFFE ID you have chosen.
+
+Apply this role to your Teleport cluster using `tctl`:
+
+```code
+$ tctl create -f role.yaml
+```
+
+You now need to assign this role to the bot:
+
+```code
+$ tctl bots update my-bot --add-roles my-workload-aws
+```
+
+## Step 3/4. Issue Workload Identity JWTs
+
+You'll now configure `tbot` to issue and renew the short-lived JWT SVIDs for
+your workload. It'll write the JWT as a file on disk, where you can then
+configure AWS clients and SDKs to read it.
+
+Take your already deployed `tbot` service and configure it to issue SPIFFE SVIDs
+by adding the following to the `tbot` configuration file:
+
+```yaml
+outputs:
+  - type: spiffe-svid
+    destination:
+      type: directory
+      path: /opt/workload-identity
+    svid:
+      path: /svc/example-service
+    jwts:
+      - audience: sts.amazonaws.com
+        file_name: aws-jwt
+```
+
+Replace:
+
+- /opt/workload-identity with the directory where you want the JWT to be
+  written.
+- /svc/example-service with the SPIFFE ID you have chosen.
+
+Restart your `tbot` service to apply the new configuration. You should see a
+file created at `/opt/workload-identity/aws-jwt` containing the JWT.
+
+## Step 4/4. Configure AWS CLIs and SDKs
+
+Finally, you need to configure the AWS CLIs and SDKs to use the JWT SVID for 
+authentication.
+
+This can be done using the configuration file located at `~/.aws/config` or by
+using environment variables.
+
+To proceed, you'll need to know the ARN of the role you created earlier.
+
+<Tabs>
+<TabItem label="Configuration File">
+
+Add the following to your `~/.aws/config` file:
+
+```ini
+# You can replace "workload-id-demo" with a recognisable name that identifies
+# your use-case.
+[profile workload-id-demo]
+# Replace with the ARN of the role you created earlier.
+role_arn=arn:aws:iam:123456789012:role/workload-id-demo
+# Replace with the directory and file name you configured `tbot` to write the
+# JWT to.
+web_identity_token_file=/opt/workload-identity/aws-jwt
+```
+
+</TabItem>
+<TabItem label="Environment Variables">
+
+Configure the following environment variables:
+
+- `AWS_ROLE_ARN`: The ARN of the role you created earlier, e.g 
+  `arn:aws:iam::123456789012:role/workload-id-demo`
+- `AWS_WEB_IDENTITY_TOKEN_FILE`: The path to the JWT file that `tbot` is writing,
+  e.g `/opt/workload-identity/aws-jwt`
+
+</TabItem>
+</Tabs>
+
+You can now test authenticating to the AWS S3 API. Create a file which you can
+upload to your bucket:
+
+```code
+$ echo "Hello, World!" > hello.txt
+```
+
+Now, use the AWS CLI to upload this file to your bucket:
+
+```code
+$ aws s3 cp hello.txt s3://workload-id-demo
+```
+
+If everything is configured correctly, you should see this file uploaded to your
```

Comment:
While a user should know how to check that the file is in the bucket, it would be a nice convenience step to add an `aws s3` command that the user could copy/paste.

Reply:
a29d4c92ad